PrebidMobile/API1.0/src/main/java/org/prebid/mobile/AdUnit.java

```diff
@@ -107,7 +115,11 @@ public void fetchDemand(@NonNull Object adObj, @NonNull OnCompleteListener liste
         }
         if (Util.supportedAdObject(adObj)) {
             fetcher = new DemandFetcher(adObj);
-            RequestParams requestParams = new RequestParams(configId, adType, sizes, keywords);
+
+            HashMap<String, Object> additionalMap = new HashMap<>(1);
+            additionalMap.put(RequestParams.INSTL_MIN_SIZE_PERC_KEY, minSizePerc);
+
+            RequestParams requestParams = new RequestParams(configId, adType, sizes, keywords, additionalMap);
```

Comment:
Why do we need an `additionalMap` for passing the `minSizePerc` when currently we have only one object to pass to the `RequestParams`? This HashMap is certainly introducing overheads as well.

Reply:
Extensibility. I would like to eliminate passing a lot of optional arguments into constructor because it creates unnecessary state and makes initializing process overhead. I am sure that the `RequestParams` class will be extended in a future by adding new features

By the way, I have a plan to improve this logic soon and provide you a PR

Reply:
Hi @yoalex5,

I am not in support of having HashMap for just one parameter. But yes, you can create a PR for refactoring the RequestParams as a whole and maybe have all the parameters attached to the HashMap. Or you can come up with a better approach.

Reply:
I agree with you.

Reply:
@avohraa, are you going to approve this PR or you are waiting for changes from my side ?

Reply:
I am still waiting for your changes.

Reply:
okay. I will do